src/GraphResponseHandler.ts

```diff
@@ -149,16 +154,44 @@ export class GraphResponseHandler {
 						 *
 						 *  So assuming it as a stream type so returning the body.
 						 */
+						shouldConsumeResponse = false;
 						responseValue = Promise.resolve(clonedRawResponse.body);
 					}
 					break;
 			}
+			if (shouldConsumeResponse) {
+				GraphResponseHandler.consumeResponse(rawResponse);
+			}
+			if (responsePromise && !responseValue) {
+				responseValue = await responsePromise;
+			}
 		} catch (error) {
 			throw error;
+		} finally {
+			if (shouldConsumeResponse) {
+				GraphResponseHandler.consumeResponse(clonedRawResponse);
+			}
 		}
 		return responseValue;
 	}
 
+	/**
+	 * @private
+	 * @static
+	 * Consume the response, so that we don't hit the highWaterMark in Node.js
+	 * @param {Response} rawResponse - The response object
+	 */
+	private static consumeResponse(rawResponse: Response): void {
+		if (rawResponse.bodyUsed) {
+			return;
+		}
+
+		const body = rawResponse.body;
+		if (typeof (body as any).resume === "function") {
```

Comment:
There are two ways of making requests with this library one is by passing in the callback and another one is not passing anything will return a promise.
In the first case - error, parsed response and raw response are the three arguments that will be passed to the callback. in this case, the user will be expecting the raw untouched response. This is the reason that we clone the raw response is to make sure we pass in the unread raw response to the callback. if we consume the response body, the user will be receiving the response with the already ready body.

Reply:
The main culprit for this problem is cloning the response and the reason for the same is to facilitate passing raw response to the callback. We have an alternate way for the user to receive the raw response (see [getting a raw response](https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/dev/docs/GettingRawResponse.md)). So we can avoid this cloning of raw response completely which eliminates the problem. Yes, this would be a breaking change, the good part is we have v2.0 lined up for the preview. so we can add this to v2.0.